packages/react-vapor/src/components/modal/tests/ConfirmationModalProvider.spec.tsx

```diff
@@ -1,155 +1,134 @@
-            confirmationModalProvider.unmount();
-        }
-    });
-
-    const shallowMountUnsavedModalProvider = (shouldConfirm: boolean = false) =>
-        shallow(
-            <ConfirmationModalProvider
-                shouldConfirm={shouldConfirm}
-                modalTitle="Are you sure?"
-                modalBodyChildren={<div>some content</div>}
-            >
-                {({promptBefore}) => (
-                    <ModalCompositeConnected
-                        id="👾"
-                        title="🙄"
-                        classes={['mod-slide-in-bottom', 'mod-big', 'mod-stick-bottom']}
-                        modalBodyChildren="🙈 🙉 🙊"
-                        modalFooterChildren={
-                            <button
-                                className="btn"
-                                onClick={() => promptBefore(promptBeforeClickActionSpy) && regularClickActionSpy()}
-                            />
-                        }
-                        modalBodyClasses={['mod-header-padding', 'mod-form-top-bottom-padding']}
-                        docLink={{url: 'https://www.coveo.com', tooltip: {title: 'Go to coveo.com'}}}
-                    />
-                )}
-            </ConfirmationModalProvider>
-        );
-
-    const heavilyMountUnsavedModalProvider = (shouldConfirm: boolean = false) =>
-        mount(
-            <ConfirmationModalProvider
-                shouldConfirm={shouldConfirm}
-                modalTitle="Are you sure?"
-                modalBodyChildren={<div>some content</div>}
-            >
-                {({promptBefore}) => (
-                    <Provider store={store}>
+    describe('when shouldConfirm prop is true:', () => {
+        it('should open the ConfirmationModal to interrupt the leaving action if isDirty is true', () => {
+            renderModal(
+                <ConfirmationModalProvider
+                    shouldConfirm
+                    confirmationModalId="👾-unsaved-modal"
+                    modalTitle="Are you sure?"
+                    modalBodyChildren="Big bada boom"
+                >
+                    {({promptBefore}) => (
                         <ModalCompositeConnected
+                            isDirty
                             id="👾"
                             title="🙄"
                             classes={['mod-slide-in-bottom', 'mod-big', 'mod-stick-bottom']}
-                            modalBodyChildren="🙈 🙉 🙊"
+                            modalBodyChildren="Hakuna matata"
                             modalFooterChildren={
-                                <button
-                                    className="btn"
-                                    onClick={() => promptBefore(promptBeforeClickActionSpy) && regularClickActionSpy()}
-                                />
+                                <button className="btn" onClick={() => promptBefore(jest.fn())}>
+                                    here
```

Comment:
We should probably find a better name for that button.

```suggestion
                                    prompt confirmation modal
```